### PR TITLE
S1854: Add FP repro for #9466, #9467, #9468

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Reproduces  #9466, #9467, #9468

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs [hunk 1]
```diff
@@ -1516,3 +1516,114 @@ public void TryReturns_Loop()
...
+            CanThrow();
+            if (condition)
+            {
+                CanThrow();
+            }
+            value = 200;
+        }
+        catch (Exception exc)
+        {
+        }
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1558]: Slightly more ecological.
```suggestion
        catch (Exception ex) {  }
```

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs [hunk 2]
```diff
@@ -1516,3 +1516,114 @@ public void TryReturns_Loop()
...
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1577]:
```suggestion
        catch (Exception ex) { }
```

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs [hunk 3]
```diff
@@ -1516,3 +1516,114 @@ public void TryReturns_Loop()
...
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/9468
+    public void NestedCatchAndRethrow()
+    {
+        var value = 100;
+        try
+        {
+            try
+            {
+                CanThrow();
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1611]: https://i.imgflip.com/7uu6kj.jpg

**Review by mary-georgiou-sonarsource — APPROVED** [review]
minor comments

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1558]: While I'd normally do that, in this case, the catch has significant impact on the flow so I wanted to waste a bit of space on it.

**Pull request merged**